### PR TITLE
Add Support for Custom Headers

### PR DESCRIPTION
**Pull request by bardabun**
This PR introduces support for adding custom headers to the Fluent Bit Logz.io output plugin. Changes include:

- Addition of SetHeaders function to configure custom headers.
- Modification of client.go to include custom headers in HTTP requests.
- Updates to client_test.go to test custom headers functionality.

### CODE REVIEW

File: output/out_logzio_test.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func TestSerializeRecord(test *testing.T) {
 	record["five"] = 5
 	testServer := time.Now()
 
-	serialize, err := serializeRecord(testServer, "atag", record, "logzio", defaultId)
+	serialize, err := serializeRecord(testServer, "atag", record, "logzio", defaultId, false, false, "_")
```
Comment (yotamloe) [inline output/out_logzio_test.go:26]: Why do we need to change the parameters here?

File: output/client_test.go [hunk 2]
```diff
@@ -28,11 +28,20 @@ func TestRequestHeaders(test *testing.T) {
...
 		require.Equal(test, "gzip", r.Header.Get("Content-Encoding"))
 		require.Equal(test, "application/json", r.Header.Get("Content-Type"))
+		require.Equal(test, "header_value_1", r.Header.Get("X-Custom-Header1"))
+		require.Equal(test, "header_value_2", r.Header.Get("X-Custom-Header2"))
 		w.WriteHeader(http.StatusOK)
 	}))
 	defer testServer.Close()
 
-	logzioClient := LogzioTestClient(testServer.URL)
+	headers := map[string]string{
```
Comment (yotamloe) [inline output/client_test.go:37]: I think we need coverage for more cases like:
1. Test with No Headers
2. Test with Duplicate Headers
3. Test with Malformed Header String (missing `:`)
4. Test with Mixed Valid and Invalid Headers

File: output/out_logzio.go [hunk 3]
```diff
@@ -229,10 +231,22 @@ func initConfigParams(ctx unsafe.Pointer) error {
 	proxyUser := output.FLBPluginConfigKey(ctx, "proxy_user") // admin
 	proxyPass := output.FLBPluginConfigKey(ctx, "proxy_pass") // password1234
 
+	headers := make(map[string]string)
+	headerConfig := output.FLBPluginConfigKey(ctx, "headers")
+	if headerConfig != "" {
+		for _, header := range strings.Split(headerConfig, ",") {
```
Comment (yotamloe) [inline output/out_logzio.go:237]: I think we need some validation for the header structure here for cases where the headers string is malformed like missing `:`  or for cases where we cant parse it correctly and add a warning log

**Review by yotamloe — CHANGES_REQUESTED** [review]
looks good, see my comments and please update the readme with the new options and add an changelog entry

File: output/out_logzio_test.go (hunk 1 above)
Comment (bardabun) [reply output/out_logzio_test.go:26]: I changed the `serializeRecord` function call in the tests to include additional parameters (`dedotEnabled`, `dedotNested`, `dedotNewSeperator`) to match the updated function signature. The tests were initially broken due to missing these parameters, and I fixed them by aligning with the new definition. These parameters control dedotting behavior, ensuring field names are appropriately processed.

File: output/out_logzio.go (hunk 3 above)
Comment (bardabun) [reply output/out_logzio.go:237]: I updated the initConfigParams function to handle header configuration more robustly. The updated function now parses the headerConfig string, checking for correct key-value formatting. It logs warnings for malformed headers, headers without keys, and duplicate keys, where the latter will overwrite the previous value.

File: output/client_test.go (hunk 2 above)
Comment (bardabun) [reply output/client_test.go:37]: I added no header test.
About the duplication scenario, a headers map will only store the last value for a given header key if there are duplicates. Go maps have unique keys, so if a user sets duplicate headers like X-Custom-Header1: header_value_1, X-Custom-Header1: header_value_2, only header_value_2 will be retained. The logging I added to 'initConfigParams' func in 'out_logzi.go' will notify the user of any duplicate keys being overwritten.

**Review by yotamloe — APPROVED** [review]

**Pull request merged**